basil/utils/utils.py

```diff
@@ -41,7 +41,7 @@ def bitarray_to_byte_array(bitarr):
     ba = bitarray(bitarr, endian=bitarr.endian())
     ba.reverse()  # this flip the byte order and the bit order of each byte
     bs = np.frombuffer(ba.tobytes(), dtype=np.uint8)  # byte padding happens here, bitarray.tobytes()
-    bs = (bs * 0x0202020202 & 0x010884422010) % 1023
+    bs = (bs.astype(np.uint64) * 0x0202020202 & 0x010884422010) % 1023
```

Comment:
I would argue it is a bit easier to follow if you do `bs * np.uint64(0x0202020202)` to make the casting more explicit and keep `bs` an array of `uint8` aka bytes. Performance seems to be about the same with a very simple test.

Reply:
Ok, I agree.